src/CoreWCF.NetTcp/tests/ServiceContract/ITestService.cs

```diff
@@ -42,5 +42,13 @@ public interface ITestService
         [OperationContract(Name = "EchoForImpersonation", Action = Constants.OPERATION_BASE + "EchoForImpersonation",
         ReplyAction = Constants.OPERATION_BASE + "EchoForImpersonationResponse")]
         string EchoForImpersonation(string echo);
+
+        [OperationContract(Name = "EchoForAuthorizarionOneRole", Action = Constants.OPERATION_BASE + "EchoForAuthorizarionOneRole",
+        ReplyAction = Constants.OPERATION_BASE + "EchoForAuthorizarionOneRoleResponse")]
+        string EchoForAuthorizarionOneRole(string echo);
+
+        [OperationContract(Name = "EchoForAuthorizarionNoRole", Action = Constants.OPERATION_BASE + "EchoForAuthorizarionNoRole",
+        ReplyAction = Constants.OPERATION_BASE + "EchoForAuthorizarionNoRoleResponse")]
+        string EchoForAuthorizarionNoRole(string echo);
     }
```

Comment:
These appear to match exactly with ClientContract/ITestServices.cs, can we derive a base class to derive from to eliminate the same code or do you foresee these diverging in the future?

Reply:
that's right Jonathan ! But the objective here is to recreate real world scenario, where both contracts remain same across client/server... this gives us a good readability... in real world, customers just uses(mostly) the wsdl to generate client and they don't write much code for getting client contract..